Assignment_1/palindrome.js

```diff
@@ -0,0 +1,46 @@
+class Node {
+    constructor(val) {
+        this.data = val;
+        this.next = null;
+    }
+}
+//
+const palindrome = (head) => {
+    if (!head || !head.next) {
+        return true; 
+    }//empty or one letter
+
+    let fast = head, slow = head;
+    const stack = [];
+
+    while (fast && fast.next) {
+        stack.push(slow.data);
+        fast = fast.next.next;
+        slow = slow.next; 
+    }//push first half of linked list onto stack
+
+    if (fast) { // if list has odd number of elements move slow past middle
+        slow = slow.next;
+    }
+
+    //going to compare stack aka first half with rest aka slow
+    while (slow) {
+        const pop = stack.pop();
+        if (pop !== slow.data) {
```

Comment:
Because you don't refer to `pop` anywhere other than these two lines, you can condense this into one line of code.
```suggestion
        if (stack.pop() !== slow.data) {
```